app/src/main/java/com/epfl/drawyourpath/challenge/DailyGoal.kt

```diff
@@ -0,0 +1,106 @@
+package com.epfl.drawyourpath.challenge
+
+import java.time.LocalDate
+
+/**
+ * class representing the Daily Goal that the user set
+ */
+data class DailyGoal(
+    var distanceInKilometerGoal: Double,
+    var timeInMinutesGoal: Double,
+    var nbOfShapesGoal: Int,
+    var distanceInKilometerProgress: Double = 0.0,
+    var timeInMinutesProgress: Double = 0.0,
+    var nbOfShapesProgress: Int = 0,
+    val date: LocalDate = LocalDate.now()
+) : java.io.Serializable {
+
+    /**
+     * create a new Empty DailyGoal based on the current one
+     *
+     * @return a new DailyGoal
+     */
+    fun createNewGoalFromThis(): DailyGoal {
+        return DailyGoal(this.distanceInKilometerGoal, this.timeInMinutesGoal, this.nbOfShapesGoal)
+    }
+
+    /**
+     * should only be used inside [DailyGoalViewAdapter]

```

Comment:
you should add the correspondance between each value in the comments : 0-> distance , 1 -> activity time

Reply:
or you can create an enumeration like explain before

Reply:
This logic should only be useful in the class DailyGoalViewAdapter so I moved it there.